lib/AST/Stmt.cpp

```diff
@@ -299,11 +299,37 @@ ASTNode BraceStmt::findAsyncNode() {
   return asyncFinder.getAsyncNode();
 }
 
-Expr *BraceStmt::getSingleExpressionElement() const {
-  if (getElements().size() != 1)
-    return nullptr;
+static bool hasSingleActiveElement(ArrayRef<ASTNode> elts) {
+  while (true) {
+    // Single element brace.
+    if (elts.size() == 1)
+      return true;
+
+    // See if we have a #if as the first element of a 2 element brace, if so we
+    // can recuse into its active clause. If so, the second element will be the
+    // active element.
+    if (elts.size() == 2) {
+      if (auto *D = elts.front().dyn_cast<Decl *>()) {
+        if (auto *ICD = dyn_cast<IfConfigDecl>(D)) {
```

Comment:
If you wanted to, I think you can wrap this in a single `if`

```suggestion
      if (auto *D = dyn_cast_or_null<IfConfigDecl>(elts.front().dyn_cast<Decl *>())) {
```

Reply:
I think that would require a line break, so IMO is more readable as-is, as it would span two lines anyways (albeit with less indentation)

Reply:
Ah, OK. That’s good 👍🏽